### PR TITLE
Fix misc warnings and missing include files

### PR DESCRIPTION
**Pull request by ggouaillardet**
:bot:assign: @rhc54 
:bot:milestone:v2.0.0
:bot:label:Code-cleanup-low-priority

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ompi-release-pr/1396/ for details.

File: opal/runtime/opal_progress.c [hunk 1]
```diff
@@ -235,10 +239,14 @@ opal_progress_event_users_increment(void)
 void
 opal_progress_event_users_decrement(void)
 {
+#if OPAL_ENABLE_DEBUG || ! OPAL_PROGRESS_USE_TIMERS
```
Comment (jsquyres) [inline opal/runtime/opal_progress.c:242]: Is this second conditional (`!OPAL_PROGRESS_USE_TIMERS`) correct?

**Comment by jsquyres** [comment]
It seems like there are more than just warnings fixes in here, and there are some actual bug fixes, too.

File: opal/runtime/opal_progress.c (hunk 1 above)
Comment (ggouaillardet) [inline opal/runtime/opal_progress.c:242]: it looks correct to me, we only need `val` in order to print it (e.g. `OPAL_ENABLE_DEBUG`) or a few lines later if `! OPAL_PROGRESS_USE_TIMERS`

Comment (jsquyres) [inline opal/runtime/opal_progress.c:242]: ok

**Comment by jsquyres** [comment]
@rhc54 Can you have a look at this one?

**Comment by rhc54** [comment]
:+1:

**Pull request merged**